### PR TITLE
disallow extend value itself

### PR DESCRIPTION
**Pull request by b2whats**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -7,7 +8,10 @@
 export default function jssExtend() {
   function extend(rule, newStyle, style) {
     if (typeof style.extend == 'string') {
-      if (rule.options && rule.options.sheet) {
+      if (style.extend === rule.selectorText) {
+        warn(`Selector '${rule.selectorText}' extends itself`)
+      }
+      else if (rule.options && rule.options.sheet) {
         const refRule = rule.options.sheet.getRule(style.extend)
```
Comment (kof) [inline src/index.js]: better here:

if (refRule) {
  if (refRule === rule) warn(...)
  else extend(rule, newStyle, refRule.originalStyle)
}

**Comment by b2whats** [comment]
update

File: src/index.js [hunk 2]
```diff
@@ -9,7 +10,10 @@ export default function jssExtend() {
     if (typeof style.extend == 'string') {
       if (rule.options && rule.options.sheet) {
         const refRule = rule.options.sheet.getRule(style.extend)
-        if (refRule) extend(rule, newStyle, refRule.originalStyle)
+        if (refRule) {
+          if (refRule === rule) warn(`Selector '${rule.selectorText}' extends itself`)
```
Comment (kof) [inline src/index.js]: we need to use rule.name there, selector is generated in named rules.

Comment (kof) [inline src/index.js]: Also in text, we better use double quotes: `Selector "${rule.name}"` extends itself.`

Comment (kof) [inline src/index.js]: Also user won't know which sheet it is and we have no sheet names.

Comment (kof) [inline src/index.js]: Maybe like this `warn(`Rule "${rule.name}"` extends itself.`, rule)`

**Comment by kof** [comment]
merged, thanks

**Pull request merged**